card/mod/pointer/set/abstract/02_pointer/item_api.rb

```diff
@@ -0,0 +1,172 @@
+# ~~~~~~~~~~~~ READING ITEMS ~~~~~~~~~~~~
+
+# While each of the three main methods for returning lists of items can handle arguments,
+# they are most commonly used without them.
+
+# @return [Array] list of Card::Name objects
+# @param args [Hash]
+# @option args [String] :content override card content
+# @option args [String, Card::Name, Symbol] :context name in whose context relative items
+#         will be interpreted. For example. +A in context of B is interpreted as B+A
+#         context defaults to pointer card's name. If value is `:raw`, then name is not
+#         contextualized
+# @option args [String, Integer] :limit max number of cards to return
+# @option args [String, Integer] :offset begin after the offset-th item
+def item_names args={}
+  raw_item_strings(args[:content], args[:limit], args[:offset]).map do |item|
+    clean_item_name item, args[:context]
+  end.compact
+end
+
+# @return [Array] list of integers (card ids of items)
+# @param args [Hash] see #item_names
+def item_ids args={}
+  item_names(args).map { |name| Card.fetch_id name }.compact
+end
+
+# @return [Array] list of Card objects
+# @param args [Hash] see #item_names for additional options
+# @option args [String] :complete keyword to use in searching among items
+# @option args [True/False] :known_only if true, return only known cards
+# @option args [String] :type name of type to be used for unknown cards
+def item_cards args={}
+  return item_cards_search(args) if args[:complete]
+  return known_item_cards(args) if args[:known_only]
+  all_item_cards args
+end
+
+# #item_name, #item_id, and #item_card each return a single item, rather than an array.
```

Comment:
here is the new API as discussed.

Now that it's here, I'm realizing it should be generalized. Searches should have item_name, too.

And, in fact, it occurs to me that the item args API should be as general as possible and mean the same thing for searches, pointers, and whatever else.  Some things are already quite good (offset, limit...).    Others not so much; right now `item_cards type: blah` restricts the results in searches and sets the type of new cards in pointers.  I'm thinking the more general API for the pointer use case should be `item_cards new: { type: blah }`. 

`content` has a similar problem, but I'm sure we can work around it.

It's also weird that "complete" is the only thing that triggers a search query. seems like you should be able to use any WQL.

More generally, it seems like we should harmonize things like "skip_virtual" (in fetch) and "known_only" (in item_cards)>
```
# for #fetch, #item_names, etc
new: false    # only real cards (equivalent to skip virtual)
new: :virtual # only return real cards if virtual (equivalent to known_only)
new: { ... }  # return new cards with these args
```

Reply:
another thought: any time any of these plural item methods are called, we should be using MGET and loading up the cache. could reduce memcache calls quite a bit!